src/mail-app/mail/view/MailViewModel.ts

```diff
@@ -230,12 +235,21 @@ export class MailViewModel {
 	}
 
 	private async loadExplicitMailTarget(listId: Id, mailId: Id, onMissingTargetEmail: () => unknown) {
+		// shouldStop returning true guarantees we won't load anything if the mail is not in the list already, so
+		// this will be synchronous
+		const listMail = await this.listModel?.loadAndSelect(mailId, () => true)
```

Comment:
it should stop if we get called with another id

Reply:
I misread it, you are trying to select it without loading anything. So this will only select the mail if it's already in the list.